metadoc-cli/src/main/scala/metadoc/cli/MetadocCli.scala

```diff
@@ -23,7 +27,11 @@ case class MetadocOptions(
     cleanTargetFirst: Boolean = false
 )
 
-case class MetadocSite(semanticdb: Seq[AbsolutePath], index: d.Index)
+case class MetadocSite(
+    semanticdb: Seq[AbsolutePath],
+    symbols: Seq[d.Symbol],
```

Comment:
I added it to support the reference provider without storing `references` in Index. The index gets really fat quickly and it's unnecessary to download all references to all symbols on startup time. This allows us to lookup references lazily when reference provider is invoked.

Note that `Index.symbols` is still there, but I strip out the references.